tests/unit/test_charm.py

```diff
@@ -1735,479 +1728,470 @@ def test_juju_run_exec_divergence(self, _juju_version: Mock, _topology_observer:
+            assert error_message in caplog.text
 
-            self.harness.charm.remove_secret("app", "non-existing-secret")
+            harness.charm.remove_secret("app", "non-existing-secret")
             assert (
                 "Non-existing field 'non-existing-secret' was attempted to be removed"
-                in self._caplog.text
+                in caplog.text
             )
 
-            self.harness.charm.remove_secret("unit", "non-existing-secret")
+            harness.charm.remove_secret("unit", "non-existing-secret")
             assert (
                 "Non-existing field 'non-existing-secret' was attempted to be removed"
-                in self._caplog.text
+                in caplog.text
             )
 
-    @patch("charm.JujuVersion.has_secrets", new_callable=PropertyMock, return_value=True)
-    @patch_network_get(private_address="1.1.1.1")
-    @patch("charm.PostgresqlOperatorCharm._on_leader_elected")
-    @pytest.mark.usefixtures("use_caplog")
-    def test_delete_existing_password_secrets(self, _, __):
-        """NOTE: currently ops.testing seems to allow for non-leader to remove secrets too!"""
-        self.harness.set_leader(True)
-        self.harness.charm.set_secret("app", "operator-password", "somepw")
-        self.harness.charm.remove_secret("app", "operator-password")
-        assert self.harness.charm.get_secret("app", "operator-password") is None
-
-        self.harness.set_leader(False)
-        self.harness.charm.set_secret("unit", "operator-password", "somesecret")
-        self.harness.charm.remove_secret("unit", "operator-password")
-        assert self.harness.charm.get_secret("unit", "operator-password") is None
-
-        self.harness.set_leader(True)
-        with self._caplog.at_level(logging.ERROR):
-            self.harness.charm.remove_secret("app", "operator-password")
-            assert (
-                "Non-existing secret operator-password was attempted to be removed."
-                in self._caplog.text
-            )
-
-            self.harness.charm.remove_secret("unit", "operator-password")
-            assert (
-                "Non-existing secret operator-password was attempted to be removed."
-                in self._caplog.text
-            )
 
-            self.harness.charm.remove_secret("app", "non-existing-secret")
-            assert (
-                "Non-existing field 'non-existing-secret' was attempted to be removed"
-                in self._caplog.text
-            )
+@pytest.mark.parametrize("scope,is_leader", [("app", True), ("unit", True), ("unit", False)])
+@patch_network_get(private_address="1.1.1.1")
+def test_migration_from_databag(harness, _has_secrets, scope, is_leader):
+    with (
+        patch("charm.PostgresqlOperatorCharm._on_leader_elected"),
+    ):
+        rel_id = harness.add_relation(PEER, harness.charm.app.name)
+        """Check if we're moving on to use secrets when live upgrade from databag to Secrets usage."""
```

Comment:
The docstring should come before any code. The same applies to `test_migration_from_single_secret`.